internal/controller/authenticator.go

```diff
@@ -181,11 +194,27 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyFunc := http.ProxyFromEnvironment
+		proxyURL, _ := proxyFunc(&http.Request{URL: &url.URL{Host: addr}})
+		if proxyURL == nil { // No proxy configured
+			return net.Dial("tcp", addr)
+		}
+		conn, err := proxy.Dial(ctx, "tcp", addr)
+		if err != nil {
+			logger.Warn("Failed to connect via proxy; falling back to direct connection", zap.Error(err))
+			return net.Dial("tcp", addr)
+		}
+		return conn, nil
+	}
+
 	conn, err := grpc.NewClient(
 		aud,
 		grpc.WithTransportCredentials(creds),
 		grpc.WithPerRPCCredentials(oauth.TokenSource{TokenSource: tokenSource}),
 		grpc.WithKeepaliveParams(kacp),
+		grpc.WithContextDialer(proxyDialer),
```

Comment:
Why did you need to replace the default dialer?
The default dialer is documented to already support HTTPS_PROXY out of the box:
https://github.com/grpc/grpc-go/blob/master/Documentation/proxy.md
Have you tested that it works out of the box?
If it doesn't work, add a big fat comment explaining how it is failing.